### DISCUSSION TITLE
Does wire support Kotlin Multiplatform Mobile?

### DISCUSSION THREAD

**Discussion by mahabaleshwarhnr** [square/wire#2742] [Q&A]
Hi I am trying to generate model from wire. I am getting below errors.  First of all I wanna know wire support multiplatform for mobile. 

**build.gradle.kts**

```
plugins {
    alias(libs.plugins.kotlinMultiplatform)
    alias(libs.plugins.androidLibrary)
    id("com.squareup.wire") version "4.9.3"
}
```
I followed the documentation but not helping much. I am getting plugin not found error or sometimes plugin cannot be applied `:` project.  It would be helpful if you share some examples of multiplatform

**Comment by oldergod** [comment]
Would you have the exact error you're seeing? I've never seen kotlinMultiplatform and android applied at the same time, maybe that's just some missing targets?

**Comment by udev** [comment]
I had the same problem as @mahabaleshwarhnr but was able to overcome including the plugin by using the version catalog:

```
// libs.versions.toml
[versions]
wire = "4.9.11"
[plugins]
wire = { id = "com.squareup.wire", version.ref = "wire" }

// build.gradle.kts (:project)
plugins {
  alias(libs.plugins.kotlinMultiplatform)
  alias(libs.plugins.androidLibrary)
  alias(libs.plugins.wire)
}
```

Without the version catalog approach I get the same error. Unfortunately even though the plugin is now successfully included nothing is ever generated when I build the project :/

**Reply by udev** [reply]
> Unfortunately even though the plugin is now successfully included nothing is ever generated when I build the project :/

Actually, I just had to specify the source directory 🤔 

According to the docs:

> If sourcePath isn’t set, Wire will look for .proto files in src/main/proto by default.

but if I configure wire in my multiplatform project as follows:

```
wire {
  kotlin {}
}
```

wire does not generate sources. Same exact project config with this wire config:

```
wire {
  sourcePath {
    srcDir("src/main/proto")
  }
  kotlin {}
}
```

worked as expected 🤷

**Reply by oldergod** [reply]
Thank for the extra info. I'll try to have a look at it soon
